### ISSUE TITLE
Can't install salt with raet using pip -e git

### ISSUE THREAD

**Issue by ryan-lane** [saltstack/salt#18447]
It looks like it's impossible to install salt currently without zmq, if you're using pip + git. I'd really be able to make a frozen virtualenv free of m2cypto and zmq from my git fork... How do I go about this? This is what happens when I hit the current 2014.7 branch in the upstream:

'pip install --install-option="--salt-transport=raet" -e git+https://github.com/saltstack/salt@2014.7#egg=salt

...

running develop

error: error in command line: command 'develop' has no such option 'salt_transport'

----------------------------------------
Cleaning up...'

Also, it looks like the egg always has zmq/m2crypto dependencies, even if raet is used.

**Comment by jfindlay** [comment]
@ryan-lane, is this still an issue?

**Comment by jaybocc2** [comment]
still experiencing this:
```
[ jay@Jay-MBP ]  ~/repos/salt  [530]
→  ✓ → % python setup.py install --salt-transport=raet                                                       Tuesday 2015-06-30 14:25
2015.5.1
running install
error: error in command line: command 'Install' has no such option 'salt_transport'
```

```
[ jay@Jay-MBP ]  ~/repos/deploy/saltshaker  [523]
[1] → salt-prototype ✓ → % pip install --install-option="--salt-transport=raet" salt                         Tuesday 2015-06-30 14:18
/Users/jay/.pyenv/versions/saltshaker/lib/python2.7/site-packages/pip/commands/install.py:197: UserWarning: Disabling all use of wheels due to the use of --build-options / --global-options / --install-options.
  cmdoptions.check_install_build_global(options)
Collecting salt
  Using cached salt-2015.5.2.tar.gz
Collecting Jinja2 (from salt)
  Using cached Jinja2-2.7.3.tar.gz
Collecting msgpack-python>0.3 (from salt)
  Using cached msgpack-python-0.4.6.tar.gz
Collecting PyYAML (from salt)
  Using cached PyYAML-3.11.tar.gz
Collecting MarkupSafe (from salt)
  Using cached MarkupSafe-0.23.tar.gz
Collecting requests>=1.0.0 (from salt)
  Using cached requests-2.7.0.tar.gz
Skipping bdist_wheel for salt, due to binaries being disabled for it.
Skipping bdist_wheel for Jinja2, due to binaries being disabled for it.
Skipping bdist_wheel for msgpack-python, due to binaries being disabled for it.
Skipping bdist_wheel for PyYAML, due to binaries being disabled for it.
Skipping bdist_wheel for MarkupSafe, due to binaries being disabled for it.
Skipping bdist_wheel for requests, due to binaries being disabled for it.
Installing collected packages: MarkupSafe, Jinja2, msgpack-python, PyYAML, requests, salt
  Running setup.py install for MarkupSafe
    Complete output from command /Users/jay/.pyenv/versions/saltshaker/bin/python2.7 -c "import setuptools, tokenize;__file__='/private/var/folders/3k/gy7r_ngs3knfqh1r16zj3vdc0000gn/T/pip-build-RPH5vJ/MarkupSafe/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /var/folders/3k/gy7r_ngs3knfqh1r16zj3vdc0000gn/T/pip-Q3iZZe-record/install-record.txt --single-version-externally-managed --compile --install-headers /Users/jay/.pyenv/versions/saltshaker/include/site/python2.7/MarkupSafe --salt-transport=raet:
    usage: -c [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
       or: -c --help [cmd1 cmd2 ...]
       or: -c --help-commands
       or: -c cmd --help

    error: option --salt-transport not recognized

    ----------------------------------------
Command "/Users/jay/.pyenv/versions/saltshaker/bin/python2.7 -c "import setuptools, tokenize;__file__='/private/var/folders/3k/gy7r_ngs3knfqh1r16zj3vdc0000gn/T/pip-build-RPH5vJ/MarkupSafe/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /var/folders/3k/gy7r_ngs3knfqh1r16zj3vdc0000gn/T/pip-Q3iZZe-record/install-record.txt --single-version-externally-managed --compile --install-headers /Users/jay/.pyenv/versions/saltshaker/include/site/python2.7/MarkupSafe --salt-transport=raet" failed with error code 1 in /private/var/folders/3k/gy7r_ngs3knfqh1r16zj3vdc0000gn/T/pip-build-RPH5vJ/MarkupSafe
```

**Comment by cachedout** [comment]
Fixed by #25093
